polynomials/polyreal.11.cpp

```diff
@@ -194,6 +194,58 @@ struct poly_real {
             x *= scale;
         return rn;
     }
+
+    poly_real derive(){
+        vector<T> b;
```

Comment:
Let's preallocate the vector's buffer and avoid using `push_back`

Reply:
Also, let's create a `poly_real` object right away instead of constructing it in the return statement.